DQM/L1TMonitor/src/L1THIonImp.cc

```diff
@@ -0,0 +1,494 @@
+    for (L1GctEmCandCollection::const_iterator j = l1NonIsoEmEmul->begin(); j!= l1NonIsoEmEmul->end(); j++) {
+      if (ne->bx()==0 && j->bx()==0 && std::distance(l1NonIsoEm->begin(), ne)==std::distance(l1NonIsoEmEmul->begin(),j)){
+	DECorr_[0][4]->Fill(ne->rank(), j->rank());
+	DECorr_[1][4]->Fill(ne->regionId().ieta(),j->regionId().ieta());
+	DECorr_[2][4]->Fill(ne->regionId().iphi(),j->regionId().iphi());
+      }
+    }      
+  }
+  
+  for (L1GctHFBitCountsCollection::const_iterator hfc=l1HFCounts->begin(); hfc!=l1HFCounts->end(); hfc++){ 
+    // only plot central BX
+    if (hfc->bx()==0) {
+      // Individual ring counts
+      l1GctHFRing1TowerCountPosEta_->Fill(hfc->bitCount(0));
+      l1GctHFRing1TowerCountNegEta_->Fill(hfc->bitCount(1));
+      l1GctHFRing2TowerCountPosEta_->Fill(hfc->bitCount(2));
+      l1GctHFRing2TowerCountNegEta_->Fill(hfc->bitCount(3));
+      // Correlate positive and negative eta
+      l1GctHFRing1TowerCountPosEtaNegEta_->Fill(hfc->bitCount(0),hfc->bitCount(1));
+      l1GctHFRing2TowerCountPosEtaNegEta_->Fill(hfc->bitCount(2),hfc->bitCount(3));
+    }
+    // Occupancy vs BX
+    for (unsigned i=0; i<4; i++){
+      l1GctHFRingTowerCountOccBx_->Fill(hfc->bx(),hfc->bitCount(i));
+    }
+  }
+
+  for (L1GctHFRingEtSumsCollection::const_iterator hfs=l1HFSums->begin(); hfs!=l1HFSums->end(); hfs++){ 
+    if (hfs->bx()==0) {
+      l1GctHFRing1ETSumPosEta_->Fill(hfs->etSum(0));
+      l1GctHFRing1ETSumNegEta_->Fill(hfs->etSum(1));
+      l1GctHFRingETSum_->Fill(hfs->etSum(0)+hfs->etSum(1));
+      l1GctHFRingETDiff_->Fill(abs(hfs->etSum(0)-hfs->etSum(1)));
+      if (hfs->etSum(1)!=0) l1GctHFRingRatioPosEta_->Fill((hfs->etSum(0))/(hfs->etSum(1)));
+      l1GctHFRing1PosEtaNegEta_->Fill(hfs->etSum(0),hfs->etSum(1));
+      std::vector<int> bit = SortMinBiasBit(hfs->etSum(2),hfs->etSum(3));
+      for(std::vector<int>::const_iterator it=bit.begin(); it !=bit.end(); it++){
+	l1GctMinBiasBitHFEt_->Fill(it - bit.begin(),*it);
+      }
+    }
+    for (unsigned i=0; i<4; i++){
+      l1GctHFRingETSumOccBx_->Fill(hfs->bx(),hfs->etSum(i));
+    }
+    for (L1GctHFRingEtSumsCollection::const_iterator j=l1HFSumsEmul->begin(); j!=l1HFSumsEmul->end(); j++){
+      if (hfs->bx()==0 && j->bx()==0 && std::distance(l1HFSums->begin(),hfs)==std::distance(l1HFSumsEmul->begin(),j)){
+	centralityCorr_->Fill(hfs->etSum(0),j->etSum(0));
+	centralityExtCorr_->Fill(hfs->etSum(1),j->etSum(1));
+	std::vector<int> dbit = SortMinBiasBit(hfs->etSum(2),hfs->etSum(3));
+	std::vector<int> ebit = SortMinBiasBit(j->etSum(2),j->etSum(3));
+      }
+    }
+  }
+
+  for (L1CaloRegionCollection::const_iterator it=rgn->begin(); it!=rgn->end(); it++){
+    if (it -> bx()==0){
+      int totm = 0;
+      int totp = 0;
+      int tot = 0;
+      if (it->gctEta() < 4){
+	totm += it->et();
+      }
+      if (it->gctEta() > 17){
+	totp += it->et();
+      }
+      if (it->gctEta() < 4 && it->gctEta() > 17){
+	tot += it->et();
```

Comment:
static analyzer points out that this is useless: indeed, you can just add totm and totp and drop this if and tot.